plugins/woocommerce/tests/e2e-pw/tests/merchant/create-simple-product.spec.js

```diff
@@ -57,7 +63,33 @@ test.describe.serial( 'Add New Simple Product Page', () => {
 		} );
 		await page.locator( '#title' ).fill( virtualProductName );
 		await page.locator( '#_regular_price' ).fill( productPrice );
+		await page.locator( '#_sale_price' ).fill( salePrice );
 		await page.locator( '#_virtual' ).click();
+
+		// Fill in a product description
+		await page
+			.frameLocator( '#content_ifr' )
+			.locator( '.wp-editor' )
+			.fill( productDescription );
```

Comment:
It will also be good to use user-facing locators such as any of the `page.getBy*` ones, since this is one of the best practices that Playwright suggests [here](https://playwright.dev/docs/best-practices#prefer-user-facing-attributes-to-xpath-or-css-selectors).

As a tip, you can let Playwright pick the locator for you using [UI mode](https://playwright.dev/docs/test-ui-mode#opening-ui-mode), specifically its [locator picker](https://playwright.dev/docs/test-ui-mode#pick-locator). Although not entirely perfect, the locator picker defaults to giving you a user-facing locator you can use.

Reply:
Thanks for this, @rodelgc !! It took me a while to understand how PW UI mode could help but, once I've got it, I've been able to use the suggestions to replace a bunch od locators with `getBy` instead. Please let me know what do you think about these changes when you can.